loader/loader.go

```diff
@@ -302,6 +302,25 @@ func Paths(path string, recurse bool) (paths []string, err error) {
 	return paths, err
 }
 
+// Dirs converts all filepaths to directories to use in the file watcher.
+// Some types of events will not register on the files themselves, so
+// watch the directory to prevent this problem
+func Dirs(paths []string) []string {
+	unique := map[string]struct{}{}
+
+	for _, path := range paths {
+		// TODO: /dir/dir will register top level directory /dir
```

Comment:
What is the action for the TODO here? Should it _not_ return the parent directory?

Reply:
If you say watch a directory, which is supported today. The logic here will watch the parent directory instead. It may be fine, I'll convert this to a comment instead of a TODO.